src/rqt_image_view/ratio_layouted_frame.cpp

```diff
@@ -136,6 +158,7 @@ void RatioLayoutedFrame::paintEvent(QPaintEvent* event)
   qimage_mutex_.lock();
   if (!qimage_.isNull())
   {
+    setAspectRatio(qimage_.width(), qimage_.height());
```

Comment:
Since the paint event can happen multiple times for the same image I don't think this command should be repeated every time but only be done once when a new image is received.